src/xdsl/dialects/linalg.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+from xdsl.ir import *
+from xdsl.irdl import *
+from xdsl.util import *
+from xdsl.dialects.memref import *
+from xdsl.dialects.builtin import IntegerType, Float32Type, IntegerAttr, FlatSymbolRefAttr
+from xdsl.parser import Parser
+
+
+@dataclass
+class Linalg:
+    ctx: MLContext
+
+    def __post_init__(self):
+        self.ctx.register_op(Generic)
+        self.ctx.register_op(Yield)
+        self.ctx.register_op(Fill)
+        self.ctx.register_op(Index)
+
+
+@irdl_op_definition
+class Generic(Operation):
+    name: str = "linalg.generic"
+    inputs = VarOperandDef(AnyAttr)
+    outputs = VarOperandDef(AnyOf([MemRefType, TensorType]))
+    indexing_maps = AttributeDef(ArrayAttr)
+    iterator_types = AttributeDef(ArrayAttr)
+    doc = AttributeDef(StringAttr)
+    library_call = AttributeDef(StringAttr)
+
+    region = RegionDef()
+    output = VarResultDef(TensorType)
+
+    @staticmethod
+    def parse(parser: Parser) -> Generic:
+
+        attributes = parser.parse_op_attributes()
+        parser.parse_string("ins")
+        operands = parser.parse_operands()
+        parser.parse_string("outs")
+        operands.extend(parser.parse_operands())
+
+        attributes | parser.parse_op_attributes()
```

Comment:
I think I would prefer if you would write `if attributes is not None: ...` instead, to make this a bit more clear.